### PR TITLE
Initial tests for authorize:authorize

### PR DESCRIPTION
**Pull request by pradtke**
Let me know if you have a recommended way for testing code that calls out `HTTP::redirectTrustedURL`. I avoided that code path in those tests.  For our internal modules we use the technique outlined in https://github.com/cirrusidentity/simplesamlphp-test-utils#mock-redirects

### CODE REVIEW

**Comment by tvdijen** [comment]
I think we use the builtin webserver and test for a HTTP/302 like we do here:
https://github.com/simplesamlphp/simplesamlphp/blob/ad1184b203b31e61c8b93e4dff3ad7f2653e9bc9/tests/www/IndexTest.php#L102

One remark I have is that PSR-2 mandates one class per file, so if you could split the file in two that'd be great

File: tests/modules/authorize/lib/Auth/Process/AuthorizeTest.php [hunk 1]
```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * Test for the authorize:Authorize authproc filter.
+ */
+
+namespace SimpleSAML\Test\Module\consent\Auth\Process;
```
Comment (tvdijen) [inline tests/modules/authorize/lib/Auth/Process/AuthorizeTest.php]: consent?

Comment (pradtke) [reply tests/modules/authorize/lib/Auth/Process/AuthorizeTest.php]: Ooops! I was copying the namespace form another file. I ran into another problem, after moving the Authorize class into it's own file I get this error `Class 'SimpleSAML\Test\Module\authorize\Auth\Process\TestableAuthorize' not found`   I'm guessing the autoloader doesn't know how to load test classes that follow the module namespace convention. I'm not sure how to address that: Make `_autoload_modules.php` aware of the `SimpleSAML\Test` namespace? Add a dev autoloader to composer? Or am I completely off base and the cause it something else?

Comment (tvdijen) [reply tests/modules/authorize/lib/Auth/Process/AuthorizeTest.php]: I'd say the autoload-dev option would be the nicest way to fix this. I'm surprised we don't have that in place yet to be honest.

Comment (pradtke) [reply tests/modules/authorize/lib/Auth/Process/AuthorizeTest.php]: I've added the `autoload-dev` option

**Comment by dbollaer** [comment]
Hello @pradtke ,
I encountered similar problem with my modules.
I think of using a custom loader (_autoload_modules.php), 
You should be able to fix it with:
>  "SimpleSAML\\Test\\Module\\authorize\\": "tests/lib/"

**Comment by pradtke** [comment]
@tvdijen Any chance this can be merged in soon? I have some other work I want to do in the authorize module and I'd like to have some base line unit tests committed first.

File: modules/authorize/docs/authorize.md [hunk 2]
```diff
@@ -50,14 +50,17 @@ Note: If regex is enabled, you must use the preg_match format, i.e. you have to
...
+
 ```php
 'authproc.sp' => [
     60 => [
         'class' => 'authorize:Authorize',
         'uid'   =>  [
-            '/.*@example.com/',
-            '/(user1|user2|user3)@example.edu/',
+            '/^.*@example.com$/',
+            // Use anchors to prevent matching 'wronguser1@exampled.edu.attacker.com'
```
Comment (thijskh) [inline modules/authorize/docs/authorize.md]: s/exampled/example/

File: tests/_autoload_modules.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        return; // it can't be a module test class
+    }
+    if (array_shift($elements) !== 'SimpleSAML') {
+        return; // the first element is not "SimpleSAML"
+    }
+    if (array_shift($elements) !== 'Test') {
+        return; // the second element is not "test"
+    }
+    if (array_shift($elements) !== 'Module') {
+        return; // the second element is not "module"
```
Comment (thijskh) [inline tests/_autoload_modules.php]: third

**Comment by tvdijen** [comment]
It’s done! Thanks @pradtke

**Comment by tvdijen** [comment]
Oh, I missed Thijs’ comments.. mobile app doesn’t show them.. I’ll take care of them!

**Pull request merged**